cabal-install/src/Distribution/Client/CmdBench.hs

```diff
@@ -96,6 +101,10 @@ benchAction flags@NixStyleFlags {..} targetStrings globalFlags = do
                ++ "You may wish to use 'build --only-dependencies' and then "
                ++ "use 'bench'."
 
+            fullArgs <- getFullArgs
+            when (occursOnlyOrBefore fullArgs "+RTS" "--") $
```

Comment:
I'm probably misunderstanding, but does it search for a tandalone "--" in a `cabal bench` commandline? IIRC, "--" is completely ignored by `bench` and `test`, so that wouldn't make sense, would it?

Reply:
Yeah that was why I was wondering in exactly which case we want to give this warning for `bench` and `test`. The `occursOnlyOrBefore` function worked nicely for `run` as you have either `+RTS ... --` (wrong) or `-- ... +RTS` (right). With `bench` and `test`, do we want to give a warning simply when `+RTS` occurs as a standalone argument, like `cabal test +RTS`? Then this works, but I could also just make it `elem`. 

It is more difficult to check whether something like `--test-options="+RTS"` is passed, because it is treated as one argument, so you can't match the string easily. So perhaps for `bench` and `test` the best thing is just to check if there is a standalone `+RTS`. There is the edge case that a user passes something like `cabal test --test-options="+RTS" +RTS ...` and actually wants to do this, then it would still give the warning even though the user is passing RTS options to the test-suite as well.

Reply:
I think using `elem` would be the best. That the warning then also happens when users use `cabal test --test-options="+RTS" +RTS ...` is not that bad in my opinion.

Reply:
I think warning for a standalone RTS is fine. I'd just change the warning text for test/build not to say "RTS is standalone instead of done properly", but "RTS found is standalone, which may be intentional to affect cabal, but please note RTS inside proper options is still required for RTS, if your goal is to affect the tested binary". The user may say "doh, silly computer, that's precisely why I keep the other RTS inside proper options", but I hope that's not too confusing.

Reply:
"Some RTS options were found standalone, which affect cabal instead of the binary. Please note RTS should be inside proper options if your goal is to affect the tested binary. For example, use 'cabal test --test-options="+RTS -N" to pass the '-N' RTS option to your binary."
How about this?

Reply:
Works for me. How about "Please note RTS inside proper options suffices..." to also unconfuse a user that thinks both are needed? I'm not sure how realistic such a confusion is, but one should never underestimate users. This has the extra advantage that the user doesn't get an impression that the RTS he's put inside proper options was somehow overlooked by cabal and perhaps is invisible to cabal. In other words, we pretend we see the RTS and just make sure the *other* RTS is not put there due to the confusion that both are needed.